script.js

```diff
@@ -58,20 +73,19 @@ function showCurrent(cur) {
 }
 
 function slide(num) {
-    var i;
     var pics = document.getElementsByClassName("slide");
     var dots = document.getElementsByClassName("dot");
     if (num > pics.length) {
         cur_slide = 1;
     } else if (num < 1) {
         cur_slide = pics.length;
     }
-    for (i = 0; i < pics.length; i += 1) {
-        pics[i].style.display = "none";
+    for (const pic of pics) {
+        pic.style.display = "none";
     }
-    for (i = 0; i < dots.length; i += 1) {
-        dots[i].className = dots[i].className.replace(" active", "");
+    for (const dot of dots) {
+        dot.classList.toggle('active')
     }
```

Comment:
You can manipulate an element's class directly, but using its `[classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)` object is usually easier.